src/main/host/descriptor/socket/netlink.rs

```diff
@@ -554,7 +554,6 @@ impl InitialState {
                 "Attempted to bind netlink socket to an address with non-zero groups {}",
                 addr.groups()
             );
-            return Err(Errno::EINVAL.into());
```

Comment:
@stevenengler what is our policy for shadow allowing unsupported features to silently succeed? I guess probably if we don't return the error here we should at least add a comment that says in what case the error should be returned again if more support is added in future work.

Reply:
> @ppopth Is there a reason you removed this error? Does this cause some application to fail?

Yes, it doesn't work with rust-libp2p if we returns an error. The application that I implemented in https://github.com/shadow/shadow/pull/3242 doesn't work if we return an error because it actually binds to multicast groups.

> The man page says:
> 
> > Only processes with an effective UID of 0 or the CAP_NET_ADMIN capability may send or listen to a netlink multicast group.
> 
> I think it would be reasonable to return `EPERM` here? Most applications shouldn't expect to be able to use multicast groups anyways and should probably expect permission errors.

Yeah, I think Linux returns a successful one, when it tries to bind. Please try running the application in https://github.com/shadow/shadow/pull/3242.
```bash
$ strace ./build/src/target/debug/test_libp2p_ping 2>&1 | grep "bind.*AF_NETLINK"
bind(10, {sa_family=AF_NETLINK, nl_pid=0, nl_groups=0x000110}, 12) = 0
```

Reply:
Okay I've had a chance to look into this. So when the application binds to the multicast netlink address, it's essentially subscribing to network-related OS events. Since Shadow doesn't support these, I think it should return an error for unknown bit flags. But if the application is only subscribing to IP address add/remove/change events (`RTMGRP_IPV4_IFADDR` and `RTMGRP_IPV6_IFADDR`), I think it's fine to return success since we know that IP addresses are never added/removed/changed in Shadow, so we know we're handling these flags correctly (by not doing anything at all).

My proposal is to expose the `RTMGRP_*` flags from the Linux headers in linux-api. So this means adding `linux/rtnetlink.h` to the includes in `lib/linux-api/bindings-wrapper.h`, regenerating the bindings with `(cd src/lib/linux-api/ && ./gen-kernel-bindings.sh)` and then creating `lib/linux-api/src/rtnetlink.rs` which exposes these constants. Then here in `main/host/descriptor/socket/netlink.rs`, we can check if `addr.groups()` contains only these `RTMGRP_IPV4_IFADDR` (0x10) or `RTMGRP_IPV6_IFADDR` (0x100) flags, and if so return 0. This will allow the libp2p example to work, which uses `nl_groups=0x000110`. If `addr.groups()` contains other flags that we haven't investigated, then we can return an error.

@ppopth Does this seem reasonable to you?

Reply:
@stevenengler That's a great idea. Thank you for figuring out what those flags really are.